Move macOS version validation from @cloudflare/cli to miniflare (#10211)
cloudflare/workers-sdk: .changeset/silly-things-walk.md

---
"@cloudflare/cli": minor
"miniflare": patch
"wrangler": patch
"create-cloudflare": patch
---

> Add macOS version validation to prevent EPIPE errors on unsupported macOS versions (below 13.5). Miniflare fails hard while Wrangler and C3 show warnings but continue execution.

Reviewer: ```suggestion
Add macOS version validation to prevent EPIPE errors on unsupported macOS versions (below 13.5). Miniflare and C3 hard fail, Wrangler commands that do not depend on Miniflare will show a warning.
```
